### PR TITLE
simplify code

### PR DESCRIPTION
**Pull request by derekpitt**
Hi!

This is sort of going back to where this module was, but all of the file descriptor was confusing and all over the code. It's now using simple readFile and writeFile calls.

Windows was throwing a EPERM error when using ftruncate.

let me know if I missed anything!

### CODE REVIEW

**Comment by derekpitt** [comment]
This should take care of https://github.com/tkellen/js-v8flags/issues/17

File: index.js [hunk 1]
```diff
@@ -25,78 +25,48 @@ function fail (err) {
...
 }
 
+
 module.exports = function (cb) {
-  openConfig(function (err, fd) {
-    if (err) {
-      return cb(fail(err));
+  var userHome = require('user-home');
+  var configpath = path.join(userHome || os.tmpdir(), configfile);
+  fs.exists(configpath, function(e) {
```
Comment (tkellen) [inline index.js]: fs.exists is an antipattern, the file could come into existence between this call and the call to create the file.

Comment (tkellen) [inline index.js]: From the node docs:

> fs.exists() is an anachronism and exists only for historical reasons. There should almost never be a reason to use it in your own code.

> In particular, checking if a file exists before opening it is an anti-pattern that leaves you vulnerable to race conditions: another process may remove the file between the calls to fs.exists() and fs.open(). Just open the file and handle the error when it's not there.

> fs.exists() will be deprecated.

**Comment by tkellen** [comment]
I very much appreciate the PR, but this approach will not work :(.  I think it might make sense to just switch to fs.*sync calls and use process.nextTick for the callback.

File: index.js (hunk 1 above)
Comment (derekpitt) [inline index.js]: Good call! I'll update! Thanks!

**Comment by derekpitt** [comment]
Yeah, reading the note on exists, i see why it was changed..

I can change everything to to sync calls like you said above?

**Comment by tkellen** [comment]
I *think* that should work, though I wonder if multiple processes trying to do this concurrently could cause problems. I've released so many broken versions of this damned library I don't even know anymore. Let's give it a shot.

/cc @phated

**Comment by derekpitt** [comment]
cool! I'll send another PR here in a bit! Thanks!

**Comment by tkellen** [comment]
Right on! Forgive me if I'm saying something you already know, but you can force push to this branch and it'll automatically update.

**Comment by phated** [comment]
@tkellen what's the point of switching to sync?  Async didn't break the code, just weird workarounds

**Comment by tkellen** [comment]
I was originally thinking if two instances were trying to do the work at the same time the event loop would block and prevent this from happening, but I believe the real issue is two processes, in which case, we're still fucked.  This whole module is a disaster.

**Comment by phated** [comment]
@tkellen correct, it is happening from 2 processes, which would probably cause all the sync calls to throw.  There are a bunch of weird edge cases, but once we handle them all, I think this module will be solid and awesome.

**Comment by tkellen** [comment]
You're right, I'm just frustrated. The logic around creating this config file should eventually be a module itself.

**Comment by tkellen** [comment]
So, assuming we keep the current setup--why can't we truncate on windows?

**Comment by derekpitt** [comment]
On windows, I get: ```{ [Error: EPERM, ftruncate] errno: 50, code: 'EPERM' }``` in ```truncErr```

it doesn't seem to be a permissions issue, since I get the same result when running as admin..

**Comment by pranaysharmadelhi** [comment]
I am getting error on WIndows - 
Error
    at new Parser (\gulp-less\node_modules\less\lib\less\parser.js:333:27)
    at Object.less.render (\gulp-less\node_modules\less\lib\less\index.js:18:22)
    at Transform._transform (\gulp-less\index.js:38:10)
    at Transform._read (\gulp-less\node_modules\through2\node_modules\readable-st
ream\lib\_stream_transform.js:184:10)
    at Transform._write (\gulp-less\node_modules\through2\node_modules\readable-s
tream\lib\_stream_transform.js:172:12)
    at doWrite (\gulp-less\node_modules\through2\node_modules\readable-stream\lib
\_stream_writable.js:237:10)
    at writeOrBuffer (\gulp-less\node_modules\through2\node_modules\readable-stre
am\lib\_stream_writable.js:227:5)
    at Transform.Writable.write (\gulp-less\node_modules\through2\node_modules\re
adable-stream\lib\_stream_writable.js:194:11)
    at write (\gulp\node_modules\vinyl-fs\node_modules\through2\node_modules\read
able-stream\lib\_stream_readable.js:623:24)
    at flow (\gulp\node_modules\vinyl-fs\node_modules\through2\node_modules\reada
ble-stream\lib\_stream_readable.js:632:7)

**Comment by derekpitt** [comment]
i did a minimal change to the path it uses for windows and i'm not getting the error anymore..  not sure if this is the best place to store that file?

**Comment by tkellen** [comment]
I'm +1 if it works. Any idea why appveyor is still blowing up @derekpitt ?

**Comment by derekpitt** [comment]
Not sure, but I can take a look tonight!

**Comment by derekpitt** [comment]
well, turned out that my patch didn't work, not sure why it passed on my local machine those few times.

I worked on it more using ```filelock``` but that didn't work in windows either..
then, i tried ```fs-ext``` for some locking, but looking at the issues there, seems like they have windows problems too..

maybe the right thing to do for now is to revert/or bypass the truncate stuff in windows until we can figure out a solution for caching.

or... just take out the caching. is the headache of this worth the gain of a cross-process safe cache?

on the good side, the --no-exit flag runs the tests properly in appveyor :smile:

**Comment by tkellen** [comment]
Thanks a bunch for looking into this @derekpitt! I think I have this sorted. Just published 2.0.5. Would you mind testing?

**Comment by derekpitt** [comment]
yup! looks good! thanks man!

**Pull request closed without merging**